addons/netfox/rollback/rollback-synchronizer.gd

```diff
@@ -60,6 +60,13 @@ func process_settings():
 	_inputs.clear()
 	_latest_state = NetworkTime.tick - 1
 	_earliest_input = NetworkTime.tick
+	_next_full_state = NetworkTime.tick
+	
+	# Scatter full state sends, so not all nodes send at the same tick
+	if is_inside_tree():
+		_next_full_state += hash(get_path()) % maxi(1, full_state_interval)
```

Comment:
Please consider renaming this to `_next_full_state_tick` or something, because anyone landing on this codeblock, is overwhelmed by what happens here, when the variable's type is unknown.

Reply:
Also this scatter, while beneficial for bandwidth (not spiking each X ticks) is not ideal for debugging, so I suggest a non-hash value with a project setting boolean.

Reply:
You're projecting your experience onto every reader, i.e. "this piece of code is confusing to me" becomes "anyone landing on this codeblock, is overwhelmed". Not a strong argument in my judgement.

> he will instantly think the whole class is a black box
Sounds like an exaggeration to me. 

But not knowing if the next full state is an actual state or a tick is fair. 

---

> makes debugging harder, so I suggest a non-hash value with a project setting boolean, as a new condition check above this condition.

How does scatter make debugging harder? It's a consistent offset, it will always be the same for the same node. And what is your proposal, what would the project setting do? Turn off the scatter?

Reply:
Renamed variable.

Reply:
>You're projecting your experience onto every reader, i.e. "this piece of code is confusing to me" becomes "anyone landing on this codeblock, is overwhelmed". Not a strong argument in my judgement.

You are also projecting your experience onto every reader
It's just a `_tick` suffix addition which indicates its an integer

----------

>How does scatter make debugging harder? It's a consistent offset, it will always be the same for the same node. And what is your proposal, what would the project setting do? Turn off the scatter?

I was not fast enough, I deleted it for a reason, you are right, its not a worthy addition

Reply:
>How does scatter make debugging harder? It's a consistent offset, it will always be the same for the same node. And what is your proposal, what would the project setting do? Turn off the scatter?

I had deleted that part, but wasn't fast enough, you are right, it's not a worthy addition and only bloats